inspirehep/dojson/utils.py

```diff
@@ -178,3 +178,46 @@ def get_recid_from_ref(ref_obj):
     except ValueError:
         res = None
     return res
+
+RANKS_MAPPINGS = {
+    'STAFF': {},
+    'SENIOR': {},
+    'JUNIOR': {},
+    'VISITOR': {
+      'alternative_names': ['VISITING SCIENTIST'],
+    },
+    'POSTDOC': {
+      'abbreviations': ['PD']
+    },
+    'PHD': {},
+    'MASTER': {
+      'abbreviations': ['MAS', 'MS', 'MSC']
+    },
+    'BACHELOR': {
+      'abbreviations': ['BS', 'BA','BSC']
+    },
+    'UNDERGRADUATE': {
+      'abbreviations': ['UG']
+     },
+    'STUDENT': {}
+}
+
+def classify_rank(value):
+    if not value:
+        return None
+    else:
+        casted_value = value.upper().replace('.', '')
+        for rank_name, rank_mapping in RANKS_MAPPINGS.items():
+            if rank_name in casted_value:
+                return rank_name
+            else:
+                if rank_mapping.get('alternative_names'):
+                    for alternative in rank_mapping['alternative_names']:
+                        if alternative in casted_value:
+                            return rank_name
```

Comment:
Instead of having two loops here, you could just have one key, e.g. **alternative_names**, and use that since abbreviations are just an alternative name too.